packages/shared-ux/chrome/navigation/src/navigation.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiAccordion, EuiLink, EuiSpacer, EuiText } from '@elastic/eui';
+import React from 'react';
+import { ILocatorDefinition, NavigationProps, NavItem } from '../types';
+import { useServices } from './services';
+
+export const Navigation = (props: NavigationProps) => {
+  const { id, items: navItems, highlightedItem, initiallyOpenSections } = props;
+
+  const services = useServices();
+
+  const { getLocator } = services;
+
+  const handleClick = (locator: ILocatorDefinition | undefined) => () => {
+    if (locator) {
+      const locatorInstance = getLocator(locator.id); // FIXME: use local cache rather than lookup each time
+
+      if (!locatorInstance) {
+        throw new Error(`Unresolved Locator instance for ${locator.id}`);
+      }
+
+      locatorInstance.navigateSync(locator.params);
+    }
+  };
+
+  const renderItems = (items: NavItem[]) => {
+    return items.map((item, index) => {
+      const { name, id: itemId, locator } = item;
+      return (
+        <div key={itemId}>
+          <EuiText size="s">
+            <EuiLink
+              color={highlightedItem === itemId ? 'primary' : 'subdued'}
+              onClick={handleClick(locator)}
+            >
+              {name}
+            </EuiLink>
+          </EuiText>
+          {index !== items.length - 1 ? <EuiSpacer size="m" /> : null}
+        </div>
+      );
+    });
+  };
+
+  const contents = navItems?.map((section, index) => {
+    const { items } = section;
+    return (
+      <div key={section.name} id={id}>
+        <EuiAccordion
+          paddingSize="l"
+          id={id + index}
+          initialIsOpen={initiallyOpenSections?.includes(section.name)}
+          buttonContent={section.name}
+        >
+          {items ? renderItems(items) : null}
```

Comment:
Would be nice to create a component for items:

```suggestion
          {items ? items.map(item => <NavigationItem item={item} /> : null}
```